internal/agent/device/spec/spec.go

```diff
@@ -84,6 +77,12 @@ type SpecManager struct {
 	managementClient client.Management
 	bootcClient      container.BootcClient
 
+	// cached rendered versions
+	currentRenderedVersion  string
+	desiredRenderedVersion  string
+	rollbackRenderedVersion string
```

Comment:
caching this locally eliminates the need to pull the spec into memory to check the version.